### PR TITLE
Emit warning when using the Yul optimiser

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
I think this needs at least one update in the commandline test expectations.

**Comment by axic** [comment]
> I think this needs at least one update in the commandline test expectations.

Apparently not because that requires the AssemblyStack warning fix first :(

Added two Solidity specific tests however.

File: test/cmdlineTests/standard_optimizer_yul/output.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"errors":[{"component":"general","formattedMessage":"Warning: This is a pre-release compiler version, please do not use it in production.\n","message":"This is a pre-release compiler version, please do not use it in production.","severity":"warning","type":"Warning"},{"component":"general","formattedMessage":"Warning: The Yul optimiser is still experimental. Do not use it in production unless correctness of generated code is verified with extensive tests.\n","message":"The Yul optimiser is still experimental. Do not use it in production unless correctness of generated code is verified with extensive tests.","severity":"warning","type":"Warning"}],"sources":{"A":{"id":0}}}
```
Comment (chriseth) [inline test/cmdlineTests/standard_optimizer_yul/output.json]: This will fail on release.

Comment (chriseth) [reply test/cmdlineTests/standard_optimizer_yul/output.json]: Ah sorry, it won't...

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
Basically this PR now breaks Yul in Standard Compiler if the optimiser is enabled :(

**Review by chriseth — DISMISSED** [review]

**Comment by axic** [comment]
Do we want to emit the same warning when using Yul via standard json or #6368 is enough of a warning?

**Comment by chriseth** [comment]
I think we don't need another warning when using yul and the yul optimizer in combination.

**Comment by chriseth** [comment]
Rebased and fixed test expectations.

**Comment by axic** [comment]
I think it looks good. Merging it.

**Pull request merged**